### PR TITLE
Bug 1880047 - Add missing pairs of logs to NotificationRobot

### PR DESCRIPTION
**Pull request by AndiAJ**
Summary:

- Added pairs of test logs to the `NotificationRobot`
- Converted the private variables to functions so they don't get initialised

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1880047

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt [hunk 1]
```diff
@@ -29,44 +29,48 @@ class NotificationRobot {
...
+                Log.i(TAG, "clearNotifications: Trying to click device back button")
                 mDevice.pressBack()
-                Log.i(TAG, "clearNotifications: Dismiss notifications tray by clicking device back button")
+                Log.i(TAG, "clearNotifications: Clicked device back button")
             }
         }
     }
 
     fun cancelAllShownNotifications() {
+        Log.i(TAG, "cancelAllShownNotifications: Trying to canceled all system notifications")
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt:64]: "Trying to cancel"

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt [hunk 2]
```diff
@@ -29,44 +29,48 @@ class NotificationRobot {
...
-                Log.i(TAG, "clearNotifications: Verified that clear notifications button exists")
-                clearButton.click()
-                Log.i(TAG, "clearNotifications: Clicked clear notifications button")
+            if (clearButton().exists()) {
+                Log.i(TAG, "clearNotifications:The clear notifications button exists")
+                Log.i(TAG, "clearNotifications: Trying to click the clear notifications button")
+                clearButton().click()
+                Log.i(TAG, "clearNotifications: Clicked the clear notifications button")
             } else if (notificationTray().exists()) {
+                Log.i(TAG, "clearNotifications: The notifications tray exists")
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt:55]: "The notifications tray is still displayed"

**Review by sv-ohorvath — APPROVED** [review]
Just some small nits. LGTM

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt]: Done !

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt]: Done !

**Pull request merged**